### PR TITLE
Update mark command to accept score

### PR DESCRIPTION
**Pull request by PawanPatil19**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/MarkCommand.java [hunk 1]
```diff
@@ -20,14 +24,21 @@ public class MarkCommand extends Command {
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Marks the task identified by the index number in the displayed task list as done.\n"
             + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + "Example: " + COMMAND_WORD + PREFIX_TASK_INDEX + "1" + PREFIX_SCORE + "4";
```
Comment (jereldlimjy) [inline src/main/java/seedu/address/logic/commands/MarkCommand.java:27]: Missing spaces

```suggestion
            + "Example: " + COMMAND_WORD + " " + PREFIX_TASK_INDEX + "1 " + PREFIX_SCORE + "4";
```

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 2]
```diff
@@ -10,4 +10,6 @@ public class Messages {
     public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
     public static final String MESSAGE_INVALID_TASK_DISPLAYED_INDEX = "The task index provided is invalid";
+    public static final String MESSAGE_INVALID_SCORE = "The score provided is invalid";
+    public static final String MESSAGE_TASK_ALREADY_DONE = "The task is already completed";
```
Comment (jereldlimjy) [inline src/main/java/seedu/address/commons/core/Messages.java:14]: Should provide a way to amend the score of an already marked task. What if I typed in a wrong score?

File: src/main/java/seedu/address/logic/commands/UnmarkCommand.java [hunk 3]
```diff
@@ -20,7 +22,7 @@ public class UnmarkCommand extends Command {
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Unmarks the task identified by the index number in the displayed task list as not done.\n"
             + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + "Example: " + COMMAND_WORD + PREFIX_TASK_INDEX + " 1";
```
Comment (jereldlimjy) [inline src/main/java/seedu/address/logic/commands/UnmarkCommand.java:25]: Missing spaces
```suggestion
            + "Example: " + COMMAND_WORD + " " + PREFIX_TASK_INDEX + " 1";
```

File: src/main/java/seedu/address/commons/core/Messages.java (hunk 2 above)
Comment (jereldlimjy) [reply src/main/java/seedu/address/commons/core/Messages.java:14]: Unless the flow is to unmark and then mark again, which seems iffy

Comment (PawanPatil19) [reply src/main/java/seedu/address/commons/core/Messages.java:14]: There are two options to proceed with this:
- User can use the mark command to give score again; that means no exception if the task is already marked
- Create a update command to change the score if needed

Any views on this?

Comment (zhengsterz) [reply src/main/java/seedu/address/commons/core/Messages.java:14]: i think we can just use the mark command again to give another score as it reduces the amount of commands bacause to make a new command just to give a new score seems like overkill imo

File: src/main/java/seedu/address/model/task/Task.java [hunk 4]
```diff
@@ -55,6 +57,22 @@ public boolean isDone() {
...
+     *
+     * @param score Score to be assigned to the current task
+     */
+    public void setScore(Score score) {
+        this.score = score;
+    }
+
+    /**
+     * Gets the score of the current task.
+     */
```
Comment (Yaladah) [inline src/main/java/seedu/address/model/task/Task.java:71]: Missing `@return` part of java doc.

File: src/main/java/seedu/address/model/task/Task.java [hunk 5]
```diff
@@ -113,8 +131,9 @@ public String getPersonAssignedRole() {
     /**
      * Changes status of current task as done by assigning isDone as true.
      */
```
Comment (Yaladah) [inline src/main/java/seedu/address/model/task/Task.java:133]: Missing `@return` part of java doc.

File: src/main/java/seedu/address/model/task/Task.java [hunk 6]
```diff
@@ -123,6 +142,7 @@ public String mark() {
...
```
Comment (Yaladah) [inline src/main/java/seedu/address/model/task/Task.java:142]: Also missing `@return`.

File: src/main/java/seedu/address/model/task/Date.java [hunk 7]
```diff
@@ -38,9 +38,6 @@ public static boolean isValidDate(String test) {
         return test.matches(VALIDATION_REGEX);
```
Comment (Yaladah) [inline src/main/java/seedu/address/model/task/Date.java:38]: `@return` java doc missing.

**Review by Yaladah — COMMENTED** [review]
LGTM apart from the mentioned java doc issues.

**Pull request merged**